proto/api_v3/query_service.proto

```diff
@@ -40,47 +40,64 @@ message GetTraceRequest {
-// interpret it this way. For instance, in Cassandra this limits the number of _spans_
-// that match the conditions, and the resulting number of traces can be less.
-//
-// Note: some storage implementations do not guarantee the correct implementation of all parameters.
+// There is no guaranteed ordering of the returned results.
 //
+// Note: Some storage implementations do not guarantee the correct implementation of all parameters.
 message TraceQueryParameters {
+  // `service_name` filters traces generated by a specific service.
   string service_name = 1;
+
+  // `operation_name` filters traces generated by a specific operation.
   string operation_name = 2;
 
-  // Attributes are matched against Span and Resource attributes.
-  // At least one span in a trace must match all specified attributes.
+  // `attributes` contains key-value pairs where the key is the attribute name
+  // and the value is its string representation. Attributes are matched against
+  // span and resource attributes. At least one span must match all specified attributes.
   map<string, string> attributes = 3;
 
-  // Span min start time in. REST API uses RFC-3339ns format. Required.
+  // `start_time_min` is the start of the time interval for the query. Only traces with spans
+  // that started after this time will be returned.
+  //
+  // The HTTP API uses RFC-3339ns format.
+  //
+  // This field is required.
   google.protobuf.Timestamp start_time_min = 4;
 
-  // Span max start time. REST API uses RFC-3339ns format. Required.
+  // `start_time_max` is the end of the time interval for the query. Only traces with spans
+  // that started before this time will be returned.
+  //
+  // The HTTP API uses RFC-3339ns format.
+  //
+  // This field is required.
   google.protobuf.Timestamp start_time_max = 5;
 
-  // Span min duration. REST API uses Golang's time format e.g. 10s.
+  // `duration_min` is the minimum duration of the trace. Only traces with spans that lasted
+  // at least this long will be returned.
+  //
+  // The HTTP API uses Golang's time format (e.g., "10s").
   google.protobuf.Duration duration_min = 6;
 
-  // Span max duration. REST API uses Golang's time format e.g. 10s.
+  // `duration_max` is the maximum duration of the trace. Only traces with spans that lasted
+  // at most this long will be returned.
+  //
+  // The HTTP API uses Golang's time format (e.g., "10s").
   google.protobuf.Duration duration_max = 7;
 
-  // Maximum depth of search. Depending on the backend storage
-  // implementtaion this could be like a regular LIMIT clause in SQL,
-  // but not all implementations support such accuracy and for those
-  // the larger depth value simply means more traces returned.
+  // `search_depth` defines the maximum number of results to return. The behavior depends on the
+  // backend storage implementation: some treat it like an SQL `LIMIT` clause, while others may
+  // return more traces when set to higher values.
```

Comment:
> defines the maximum number of results to return

no, it doesn't. What are even "results"? Traces or spans? The name is intentionally vague, and approximately means number of traces, but subject to various implementation constraints. The only guarantee is the higher depth should return more results.